x-pack/plugin/apm-data/src/main/resources/component-templates/metrics-apm@settings.yaml

```diff
@@ -5,10 +5,12 @@ _meta:
   managed: true
 template:
   settings:
-    codec: best_compression
-    mapping:
-      # apm@settings sets `ignore_malformed: true`, but we need
-      # to disable this for metrics since they use synthetic source,
-      # and this combination is incompatible with the
-      # aggregate_metric_double field type.
-      ignore_malformed: false
+    index:
+      codec: best_compression
+      mapping:
+        # apm@settings sets `ignore_malformed: true`, but we need
+        # to disable this for metrics since they use synthetic source,
+        # and this combination is incompatible with the
+        # aggregate_metric_double field type.
+        ignore_malformed: false
```

Comment:
@axw in your comment from https://github.com/elastic/elasticsearch/pull/117448#pullrequestreview-2457622758, did you imply that you'd want this section to be removed in this PR?

Reply:
I did, but that's OK, we can handle that in a followup.